winforms/src/toga_winforms/dialogs.py

```diff
@@ -139,32 +145,52 @@ def __init__(self, interface, title, message, content, retry, on_result):
         # Add acceptance/close buttons
         if retry:
             retry = WinForms.Button()
-            retry.Left = 290
-            retry.Top = 250
-            retry.Width = 100
+            retry.Left = self.scale_in(290)
+            retry.Top = self.scale_in(250)
+            retry.Width = self.scale_in(100)
             retry.Text = "&Retry"
             retry.Click += WeakrefCallable(self.winforms_Click_retry)
 
             self.native.Controls.Add(retry)
 
             quit = WinForms.Button()
-            quit.Left = 400
-            quit.Top = 250
-            quit.Width = 100
+            quit.Left = self.scale_in(400)
+            quit.Top = self.scale_in(250)
+            quit.Width = self.scale_in(100)
             quit.Text = "&Quit"
             quit.Click += WeakrefCallable(self.winforms_Click_quit)
 
             self.native.Controls.Add(quit)
         else:
             accept = WinForms.Button()
-            accept.Left = 400
-            accept.Top = 250
-            accept.Width = 100
+            accept.Left = self.scale_in(400)
+            accept.Top = self.scale_in(250)
+            accept.Width = self.scale_in(100)
             accept.Text = "&OK"
             accept.Click += WeakrefCallable(self.winforms_Click_accept)
 
             self.native.Controls.Add(accept)
 
+        # Required for scaling
+        self.original_control_fonts = dict()
+        self.original_control_bounds = dict()
+        for control in self.native.Controls:
+            self.original_control_fonts[control] = control.Font
+            if isinstance(control, WinForms.Button):
+                self.original_control_bounds[control] = Rectangle(
+                    self.scale_out(control.Bounds.X),
```

Comment:
There should be a comment explaining why Button needs its own branch.

Reply:
Done. Thanks!